### DISCUSSION TITLE
Output overwrite after restart

### DISCUSSION THREAD

**Discussion by WayneX-Wong** [idaholab/moose#26512] [Q&A General]
Hi，

I output exodus and checkpoints of previous transient simulation, and restart the simulation with `start_time` set to `end_time` of previous simulation.

But the newly output exodus overwrites the previous results rather than to add after them.

Is there anyway to add newly calculated result after previous output?

**Comment by GiudGiud** [comment]
Hello

I dont think we do.
@roystgnr you seen that before? Any way to append exodus data after the fact?

Guillaume

**Reply by WayneX-Wong** [reply]
@GiudGiud Thanks for your reply.

I'm trying to couple my code with MOOSE framework, but wrapping it with MOOSE wrapping interface seems diffucult to me. I think that will need a comprehensive understand of MOOSE's framework.

I try to do the coupling in a file-base loose manner, which will need to restart MOOSE simulation multiple time when off-line data transfer, so I raise this question.

Back the exodus up everytime before restart would be a straight-forward way, while may need multiple backups when simulation time-steps are more.

**Reply by roystgnr** [reply]
In theory we've got support for this at the libMesh level: just set `your_exii_writer.append(true)` before doing `ExodusII_IO::write_nodal_data()` or `ExodusII_IO::write_nodal_data_discontinuous()`.

In practice?  I've never used it ... but looking now, I do see MOOSE `Exodus.C` using it for recovery, so it must be working.  Might not be too much of a hassle to support it for non-recovery Exodus writes too.

**Reply by GiudGiud** [reply]
@WayneX-Wong it does seem like a trivial change. Add a boolean parameter to the Exodus output class, and if that boolean is true then use the setter for appending.

**Reply by WayneX-Wong** [reply]
@roystgnr @GiudGiud I follow what you suggest, add a boolean parameter and use `_exodus_io_ptr->append(true)` in `Exodus::outputSetup()` just as recovering do

But it gives a 

> Warning! The Exodus file already contains information records.
> Exodus does not support writing additional records in this situation.

during run and overwrites again.

**Reply by GiudGiud** [reply]
For the warning, you need to find the section in the code that writes the input file to the Exodus output. Then when appending, you should skip over that page. 

If it is still overwriting the field data instead of appending, maybe we are still missing a call to ->append somewhere

**Reply by WayneX-Wong** [reply]
I find this warning arising at exodusII_io_helper.C with following annotation

> There may already be information records in the file (for example, if we're appending) and in that case, according to the Exodus documentation, writing more information records is not supported.

I don't know if this means I couldn't append Exodus in such case.

Thanks again for your guidance, I'll try more.

**Reply by GiudGiud** [reply]
> writing more information records is not supported.

writing fields for more timesteps should be supported

**Reply by roystgnr** [reply]
You might want to test your new append bool in the if statement in MOOSE Exodus.C that decides whether or not to call `_exodus_io_ptr->write_information_records(_input_record);`

**Reply by WayneX-Wong** [reply]
> You might want to test your new append bool in the if statement in MOOSE Exodus.C that decides whether or not to call `_exodus_io_ptr->write_information_records(_input_record);`

I  use setter of `exodus_io->append` here in and skip calling `_exodus_io_ptr->write_information_records(_input_record)` if append, the warning disappear but overwrite still

While now I found simply using `--recover` would realize appending... I've been obsessed with restarting

Thanks again for your patience @roystgnr @GiudGiud , hope I didn't take too much of your time

**Reply by GiudGiud** [reply]
I've got to work on the moose training right now but I ll create an issue for this. And I should have time end of next week. 

I think we can make it work for restart.
